src/provider/provider_coarse.c

```diff
@@ -408,10 +677,19 @@ static void coarse_memory_provider_finalize(void *provider) {
     block_t *block = coarse_provider->block_list;
     while (block) {
         block_t *to_free = block;
+        if (block->node) {
+            free_blocks_rm_node(coarse_provider->free_blocks, block->node);
+        }
+        if (block->used) {
+            allocated_blocks_rm(coarse_provider->allocated_blocks, block->data);
```

Comment:
Maybe change a name of this func to allocated_blocks_rm_node? This will be more consistent with free_blocks_rm_node name

Reply:
Done